framework/db/schema/pgsql/CPgsqlSchema.php

```diff
@@ -165,8 +165,13 @@ protected function resolveTableNames($table,$name)
 	 */
 	protected function findColumns($table)
 	{
+		$serverVersion = $this->getDbConnection()->getServerVersion();
+		$columnDefValue = version_compare($serverVersion, '12.0', '<')
+			? 'd.adsrc' : 'CAST(pg_get_expr(d.adbin, d.adrelid) AS varchar)';
```

Comment:
@bio Can you explain why the CAST is needed?

Reply:
@marcovtwout just to maek sure we always have string. `adbin` has type `pg_node_tree` and `adrelid` has type `oid` (https://www.postgresql.org/docs/12/catalog-pg-attrdef.html)

Reply:
Correct me if I'm wrong, but I don't think casting needs to apply here. The types you mension are the parameter types `pg_get_expr()`, which has a return type of `text` (https://www.postgresql.org/docs/12/functions-info.html#FUNCTIONS-INFO-CATALOG-TABLE).

Reply:
@marcovtwout yeah, pg_get_expr returns text so no need to cast to varchar.